parse.go

```diff
@@ -350,6 +350,10 @@ func (p *parser) valueArray(it item) (interface{}, tomlType) {
 		array []interface{}
 		types []tomlType
 	)
+	// Initialize to a non-nil empty slice. This makes it
+	// consistent with how S = [] decodes into a non-nil slice
+	// inside something like struct { S []string }.
+	array = []interface{}{}
```

Comment:
Could you mention your issue in this comment? Just for future reference.

Also, might as well move this up with the declaration: `array = []interface{}{}` rather than defining it and then assigning a value.